src/line.rs

```diff
@@ -1062,6 +1123,72 @@ impl<'input, Endian> LineNumberProgramHeader<'input, Endian>
         StateMachine::new(self)
     }
 
+    /// Execute the line number program, completing the LineNumberProgramHeader
+    /// and producing an array of sequences within the line number program that
+    /// can later be used with `resume_from`.
+    ///
+    /// ```
+    /// # fn foo() {
+    /// use gimli::{LineNumberProgramHeader, NativeEndian};
+    ///
+    /// fn get_line_number_program_header<'a>() -> LineNumberProgramHeader<'a, NativeEndian> {
+    ///     // Get a line number program header from some offset in a
+    ///     // `.debug_line` section...
+    /// #   unimplemented!()
+    /// }
+    ///
+    /// let header = get_line_number_program_header();
+    /// let (complete_header, sequences) = header.sequences().unwrap();
+    /// println!("There are {} sequences in this line number program", sequences.len());
+    /// # }
+    /// ```
+    pub fn sequences(self)
+                     -> parser::Result<(LineNumberProgramHeader<'input, Endian>,
+                                        Vec<LineNumberSequence<'input, Endian>>)> {
+        let mut state_machine = self.rows();
+        let mut sequence_start_addr = None;
+        let mut sequences = Vec::new();
+        loop {
+            let sequence_end_addr;
+            match state_machine.next_row()? {
+                Some((_, row)) => {
+                    if row.end_sequence() {
+                        sequence_end_addr = row.address();
+                    } else if sequence_start_addr.is_none() {
+                        sequence_start_addr = Some(row.address());
+                        continue;
+                    } else {
+                        continue;
+                    }
+                }
+                None => break,
+            };
+
+            // We just finished a sequence.
+            sequences.push(LineNumberSequence {
+                // In theory one could have multiple DW_LNE_end_sequence opcodes
+                // in a row.
+                start: sequence_start_addr.unwrap_or(0),
+                end: sequence_end_addr,
+                rest: state_machine.opcodes.clone(),
```

Comment:
This is the opcodes from after the sequence? Also, it would be nice to limit it to only the opcodes for the sequence (so reduce the size of the input buffer to cover only the sequence).